src/overlays/actors/ovl_En_Elf/z_en_elf.c

```diff
@@ -21,99 +66,1510 @@ const ActorInit En_Elf_InitVars = {
+
+        if (this->flags & FAIRY_FLAG_BIG) {
+            scale *= 2.0f;
+        }
+
+        scale *= (this->actor.scale.x * 124.99999f);
+        Matrix_MultVec3f(&zeroVec, &mtxMult);
+        Matrix_Translate(mtxMult.x, mtxMult.y, mtxMult.z, MTXMODE_NEW);
+        Matrix_Scale(scale, scale, scale, MTXMODE_APPLY);
+    }
+
+    // do not draw wings for big fairies
+    if (this->flags & FAIRY_FLAG_BIG) {
+        if (limbIndex == 4 || limbIndex == 7 || limbIndex == 11 || limbIndex == 14) {
+            *dList = NULL;
+        }
+    }
+
+    return 0;
+}
+
+void EnElf_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    f32 alphaScale;
+    s32 envAlpha;
+    EnElf* this = THIS;
+    s32 pad1;
+    Gfx* dList;
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Elf/func_80A049B4.s")
+    if ((this->unk_2A8 != 8) && !(this->flags & 8)) {
+        if (!(player->stateFlags1 & 0x100000) || (kREG(90) < this->actor.projectedPos.z)) {
+            dList = Graph_Alloc(globalCtx->state.gfxCtx, sizeof(Gfx) * 4);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Elf/func_80A04D90.s")
+            OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_elf.c", 2730);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Elf/func_80A04DE4.s")
+            func_80094B58(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Elf/func_80A04F94.s")
+            envAlpha = (this->timer * 50) & 0x1FF;
+            envAlpha = (envAlpha > 255) ? 511 - envAlpha : envAlpha;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Elf/func_80A05040.s")
+            alphaScale = this->dissapearTimer < 0 ? (this->dissapearTimer * (7.0f / 6000.0f)) + 1.0f : 1.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Elf/func_80A05114.s")
+            gSPSegment(oGfxCtx->polyXlu.p++, 0x08, dList);
+            gDPPipeSync(dList++);
+            gDPSetPrimColor(dList++, 0, 0x01, (u8)this->innerColor.r, (u8)this->innerColor.g, (u8)this->innerColor.b,
+                            (u8)(this->innerColor.a * alphaScale));
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Elf/func_80A05188.s")
+            if (this->flags & 4) {
+                gDPSetRenderMode(dList++, G_RM_PASS, G_RM_CLD_SURF2);
+            } else {
+                gDPSetRenderMode(dList++, G_RM_PASS, G_RM_ZB_CLD_SURF2);
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Elf/func_80A05208.s")
+            gSPEndDisplayList(dList++);
+            gDPSetEnvColor(oGfxCtx->polyXlu.p++, (u8)this->outerColor.r, (u8)this->outerColor.g, (u8)this->outerColor.b,
+                           (u8)(envAlpha * alphaScale));
+            oGfxCtx->polyXlu.p = SkelAnime_Draw2(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl,
+                                                 EnElf_OverrideLimbDraw, 0, &this->actor, oGfxCtx->polyXlu.p);
```

Comment:
```suggestion
                                                 EnElf_OverrideLimbDraw, NULL, &this->actor, oGfxCtx->polyXlu.p);
```
`PostLimbDraw2` function pointer